src/components/MobileControls/MobileControl/MobileControl.tsx

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React, {ReactNode, SVGProps, memo, useMemo, useState} from 'react';
+import {ChevronDown} from '@gravity-ui/icons';
+import {Button} from '@gravity-ui/uikit';
+import block from 'bem-cn-lite';
+
+import {ListItem} from '../../../models';
+import MobileControlSheet from '../MobileControlSheet/MobileControlSheet';
+
+import './MobileControl.scss';
+
+const b = block('dc-mobile-control');
+
+export interface MobileControlProps {
+    name: string;
+    title: string;
+    labelPostfix?: string;
+    Icon: (props: SVGProps<SVGSVGElement>) => React.JSX.Element;
+    item: ReactNode;
+    displayItems: ListItem[];
+    onChangeValue?: (value: any) => void;
+}
+
+const MobileControl = memo(
+    ({name, title, labelPostfix, Icon, item, displayItems, onChangeValue}: MobileControlProps) => {
+        const labelText = useMemo(
+            () => displayItems.find((displayItem) => item === displayItem.value)?.text ?? item,
+            [item, displayItems],
+        );
+
+        const [sheetIsVisible, setSheetIsVisible] = useState(false);
+
+        const onSheetOpen = () => setSheetIsVisible(true);
+        const onSheetClose = () => setSheetIsVisible(false);
+        const onItemClick = (value: any) => (onChangeValue ? onChangeValue(value) : null);
+
+        const label = (
```

Comment:
why is this element defined separately ?

Reply:
At that moment, it seemed to me the best solution in terms of readability, but now I have changed this part of the code